mu-plugins/blocks/site-breadcrumbs/src/view.js

```diff
@@ -0,0 +1,191 @@
+/**
+ * Calculates the available space in the parent container between the breadcrumb container and the menu container.
+ *
+ * @param {HTMLElement} parent The parent container.
+ *
+ * @return {number} The available space in pixels
+ */
+const getAvailableSpace = ( parent ) => {
+	const parentWidth = parent.getBoundingClientRect().width;
+	let occupiedSpace = 0;
+
+	for ( const child of parent.children ) {
+		occupiedSpace += child.getBoundingClientRect().width;
+	}
+
+	return parentWidth - occupiedSpace;
+};
+
+/**
+ * Return whether every item in the array has the class 'is-truncated'
+ *
+ * Example HTML:
+ * <span>
+ *    <a href="#" class="is-truncated" />
+ * </span>
+ *
+ * @param {Array<HTMLElement>} arr
+ * @return {boolean}
+ */
+const allTruncated = ( arr ) =>
+	arr.every( ( { firstChild: anchorElement } ) => anchorElement.classList.contains( 'is-truncated' ) );
+
+/**
+ * Return whether every item in the array has the class 'is-hidden'
+ *
+ * @param {Array<HTMLElement>} arr
+ * @return {boolean}
+ */
+const allHidden = ( arr ) => arr.every( ( crumb ) => crumb.classList.contains( 'is-hidden' ) );
+
+/**
+ * This function collapses the breadcrumbs until the available space is greater than the breakpoint.
+ *
+ * @param {Array<HTMLElement>} arr        List of breadcrumb elements.
+ * @param {HTMLElement}        container  Breadcrumb parent container
+ * @param {number}             breakpoint The breakpoint in pixels.
+ *
+ * @return {void}
+ */
+const collapseCrumbs = ( arr, container, breakpoint ) => {
```

Comment:
Name the array for what it is?
```suggestion
const collapseCrumbs = ( breadcrumbs, container, breakpoint ) => {
```